File: server/src/message.go
package main

const (
	// New - used when sending a new file
	New = "new"

	// Done - used to indicate the end of a series of messages
	Done = "done"
)

// Message - Contains information sent over a Connection
type Message struct {
	Type string
	File string
	Body string
}

// Settings - Contains information used when syncing
type Settings struct {
	Dir     string
	Filters []string
}

// SyncDir - Contains a directory sent to the server
type SyncDir struct {
	Dir string `json:"dir"`
}

// FilterList - Contains a list of filters sent to the server
type FilterList struct {
	Filters []string `json:"filters"`
}

File: client/trace/app/src/main/java/com/example/brandon/trace/Message.java
package com.example.brandon.trace;

/**
 * Message class used to send & receive information via a ControlConnection.
 */
public class Message {
    public static final String NEW = "new";
    public static final String DONE = "done";

    public String Type;
    public String File;
    public String Body;

    public Message(String Type, String File, String Body) {
        this.Type = Type;
        this.File = File;
        this.Body = Body;
    }
}
